### DISCUSSION TITLE
Question: Are the claims made on SMACC about BT valid for this implementation?

### DISCUSSION THREAD

**Discussion by Levi-Armstrong** [BehaviorTree/BehaviorTree.CPP#613] [Q&A] [answered]
The [SMACC Comparison](https://smacc.dev/smacc-vs-behavior-trees/) makes claims about orthogonality and concurrency, but looking through this repos tutorials it seems those claims may not be valid for this repos implementation.

**Comment by SteveMacenski** [comment, marked as answer]
Assuming my understand of orthogonality in the SM context is correct, we've been using orthogonality in Nav2's behavior trees using BT.CPP since day 1. We have **many** leaf clients that call independent task servers doing independent things on independent hardware (at times), which is the same thing that SMACC does and is described in their [definitions document](https://smacc.dev/orthogonals/). I can't speak to concurrancy, but I see no reason a control flow node couldn't be made to do so, but I'm not sure if BT.CPP has such a BT node example -- but I don't think there's anything preventing it.

That entire article is incredibly biased, inaccurate, and overtly aggressive. 

A poll I ran on LinkedIn (195 responses) show a distinctly different outlook. 
![image](https://user-images.githubusercontent.com/14944147/142075508-1e2a18b3-4e5b-4cbd-815a-639f5f3bcae0.png)

**Comment by facontidavide** [comment]
I am glad you asked. I agree with Steve's comment about the article being "biased, inaccurate, and overtly aggressive".

This does not mean that BT are better than SMACC (or vice-versa). 

I am not taking any stance on this matter, because I haven't had the time, patience or need to learn SMACC.

I expressed my opinion about that article here: https://discourse.ros.org/t/introducing-the-smacc-state-machine-library/14963/9?u=facontidavide

The most important point, for me, is:

### we should understand which features are "implementation dependent" and which are an intrinsic property of Behavior Trees

This is a critical distinction, because an implementation can always be improved, and this is precisely what that article fails to acknowledge.

**Comment by miccol** [comment]
hi @Levi-Armstrong,

Thank you for bringing this up.
I agree with Steve and Davide about the nature of that article. 

Like Davide, I tried to start a discussion with the author of that article [here](https://github.com/reelrbtx/SMACC/issues/17) with no success.

**Comment by Levi-Armstrong** [comment]
Everyone, thank you for the immediate response and information which confirmed my suspicion. I will leave it up to the maintainer to close since you may prefer to leave it open.

**Comment by facontidavide** [comment]
![image](https://user-images.githubusercontent.com/2822888/142423241-bd3ffb04-39ae-4167-aa06-f404903a4888.png)

![image](https://user-images.githubusercontent.com/2822888/142423541-37a1d002-0af5-4275-a038-aaf02e1c6927.png)

I think that **keeping** the strikethrough text there, instead of actually replacing it, can tell us a lot about the sincerity of those corrections and the will of the author to engage in a constructive conversation.
